sdk/resourcemanager/azure-resourcemanager-network/src/main/java/com/azure/resourcemanager/network/implementation/VirtualNetworkGatewayConnectionImpl.java

```diff
@@ -226,12 +226,12 @@ public VirtualNetworkGatewayConnectionImpl withAuthorization(String authorizatio
                         this.resourceGroupName(),
                         this.name(),
                         new ConnectionSharedKeyInner().withValue(updateSharedKey))
-                    .doOnNext(inner -> {
-                        updateSharedKey = null;
-                    })
                     .then(myManager.serviceClient().getVirtualNetworkGatewayConnections()
                         .getByResourceGroupAsync(this.resourceGroupName(), this.name())
-                        .map(innerToFluentMap(this)));
+                        .map(innerToFluentMap(this)))
+                    .doOnSuccess(connection -> {
+                       updateSharedKey = null;
+                    });
```

Comment:
Better not change to doOnSuccess. Otherwise, the `then` failure will result a updateShareKey next time.